### ISSUE TITLE
Error when installing 

### ISSUE THREAD

**Issue by hugokitano** [kogalur/randomForestSRC#54]
I tried to install this package for the first time and got this error. What should I do? Thanks.
```
> install.packages("randomForestSRC")
Installing package into ‘/usr/local/lib/R/3.6/site-library’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/src/contrib/randomForestSRC_2.9.2.tar.gz'
Content type 'application/x-gzip' length 1129203 bytes (1.1 MB)
==================================================
downloaded 1.1 MB

* installing *source* package ‘randomForestSRC’ ...
** package ‘randomForestSRC’ successfully unpacked and MD5 sums checked
** using staged installation
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking for clang option to support OpenMP... unsupported
configure: creating ./config.status
config.status: creating src/Makevars
** libs
clang -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include  -fPIC  -g -O2  -c R_init_randomForestSRC.c -o R_init_randomForestSRC.o
clang -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include  -fPIC  -g -O2  -c randomForestSRC.c -o randomForestSRC.o
randomForestSRC.c:19576:3: warning: implicit declaration of function 'omp_init_lock' is invalid in C99 [-Wimplicit-function-declaration]
  omp_init_lock(&RF_lockEnsbUpdtCount);
  ^
randomForestSRC.c:19576:18: error: use of undeclared identifier 'RF_lockEnsbUpdtCount'; did you mean 'RF_ensbUpdtCount'?
  omp_init_lock(&RF_lockEnsbUpdtCount);
                 ^~~~~~~~~~~~~~~~~~~~
                 RF_ensbUpdtCount
randomForestSRC.c:522:17: note: 'RF_ensbUpdtCount' declared here
uint            RF_ensbUpdtCount;
                ^
randomForestSRC.c:19577:18: error: use of undeclared identifier 'RF_lockPerf'
  omp_init_lock(&RF_lockPerf);
                 ^
randomForestSRC.c:19579:5: error: use of undeclared identifier 'RF_lockPartial'
    RF_lockPartial = ompvector(1, RF_observationSize);
    ^
randomForestSRC.c:19579:22: warning: implicit declaration of function 'ompvector' is invalid in C99 [-Wimplicit-function-declaration]
    RF_lockPartial = ompvector(1, RF_observationSize);
                     ^
randomForestSRC.c:19581:23: error: use of undeclared identifier 'RF_lockPartial'
      omp_init_lock(&(RF_lockPartial[i]));
                      ^
randomForestSRC.c:19599:5: error: use of undeclared identifier 'RF_lockWeight'; did you mean 'RF_crWeight'?
    RF_lockWeight = (omp_lock_t **) new_vvector(1, gMembershipSize, NRUTIL_OMPLPTR);
    ^~~~~~~~~~~~~
    RF_crWeight
randomForestSRC.c:191:11: note: 'RF_crWeight' declared here
double   *RF_crWeight;
          ^
randomForestSRC.c:19599:35: error: expected expression
    RF_lockWeight = (omp_lock_t **) new_vvector(1, gMembershipSize, NRUTIL_OMPLPTR);
                                  ^
randomForestSRC.c:19599:22: error: use of undeclared identifier 'omp_lock_t'
    RF_lockWeight = (omp_lock_t **) new_vvector(1, gMembershipSize, NRUTIL_OMPLPTR);
                     ^
randomForestSRC.c:19601:7: error: use of undeclared identifier 'RF_lockWeight'; did you mean 'RF_crWeight'?
      RF_lockWeight[i] = ompvector(1, RF_observationSize);
      ^~~~~~~~~~~~~
      RF_crWeight
randomForestSRC.c:191:11: note: 'RF_crWeight' declared here
double   *RF_crWeight;
          ^
randomForestSRC.c:19601:26: warning: implicit declaration of function 'ompvector' is invalid in C99 [-Wimplicit-function-declaration]
      RF_lockWeight[i] = ompvector(1, RF_observationSize);
                         ^
randomForestSRC.c:19605:25: error: use of undeclared identifier 'RF_lockWeight'
        omp_init_lock(&(RF_lockWeight[i][j]));
                        ^
randomForestSRC.c:19608:5: error: use of undeclared identifier 'RF_lockWeightRow'
    RF_lockWeightRow = ompvector(1, gMembershipSize);
    ^
randomForestSRC.c:19608:24: warning: implicit declaration of function 'ompvector' is invalid in C99 [-Wimplicit-function-declaration]
    RF_lockWeightRow = ompvector(1, gMembershipSize);
                       ^
randomForestSRC.c:19610:23: error: use of undeclared identifier 'RF_lockWeightRow'; did you mean 'RF_crWeight'?
      omp_init_lock(&(RF_lockWeightRow[i]));
                      ^~~~~~~~~~~~~~~~
                      RF_crWeight
randomForestSRC.c:191:11: note: 'RF_crWeight' declared here
double   *RF_crWeight;
          ^
randomForestSRC.c:19629:7: error: use of undeclared identifier 'RF_lockVimp'
      RF_lockVimp = (omp_lock_t **) new_vvector(1, xVimpSize, NRUTIL_OMPLPTR);
      ^
randomForestSRC.c:19629:35: error: expected expression
      RF_lockVimp = (omp_lock_t **) new_vvector(1, xVimpSize, NRUTIL_OMPLPTR);
                                  ^
randomForestSRC.c:19629:22: error: use of undeclared identifier 'omp_lock_t'
      RF_lockVimp = (omp_lock_t **) new_vvector(1, xVimpSize, NRUTIL_OMPLPTR);
                     ^
randomForestSRC.c:19631:9: error: use of undeclared identifier 'RF_lockVimp'
        RF_lockVimp[i] = ompvector(1, obsSize);
        ^
randomForestSRC.c:19631:26: warning: implicit declaration of function 'ompvector' is invalid in C99 [-Wimplicit-function-declaration]
        RF_lockVimp[i] = ompvector(1, obsSize);
                         ^
randomForestSRC.c:19635:27: error: use of undeclared identifier 'RF_lockVimp'
          omp_init_lock(&(RF_lockVimp[i][j]));
                          ^
randomForestSRC.c:19638:7: error: use of undeclared identifier 'RF_lockVimpRow'
      RF_lockVimpRow = ompvector(1, obsSize);
      ^
randomForestSRC.c:19638:24: warning: implicit declaration of function 'ompvector' is invalid in C99 [-Wimplicit-function-declaration]
      RF_lockVimpRow = ompvector(1, obsSize);
                       ^
randomForestSRC.c:19640:25: error: use of undeclared identifier 'RF_lockVimpRow'
        omp_init_lock(&(RF_lockVimpRow[i]));
                        ^
randomForestSRC.c:19642:7: error: use of undeclared identifier 'RF_lockVimpCol'
      RF_lockVimpCol = ompvector(1, xVimpSize);
      ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
6 warnings and 20 errors generated.
make: *** [randomForestSRC.o] Error 1
ERROR: compilation failed for package ‘randomForestSRC’
* removing ‘/usr/local/lib/R/3.6/site-library/randomForestSRC’
Warning in install.packages :
  installation of package ‘randomForestSRC’ had non-zero exit status

The downloaded source packages are in
	‘/private/var/folders/h4/sswkds452lqb6tc2zwhrcd_w0000gn/T/RtmpQKWNim/downloaded_packages’
```

**Comment by kogalur** [comment]
What platform is this?  What version OS are you running?  What version of clang is installed?

**Comment by hugokitano** [comment]
I am on Mac OS 10.13.6, and my version of Clang is "Apple LLVM version 10.0.0 (clang-1000.10.44.4)"

I tried to install on RStudio, and then tried installing from the source .tar file and got the same error.

**Comment by kogalur** [comment]
Have you tried installing the CRAN binary version for Mac OS, that is available on CRAN? You can download it and install it from the command line.  That is the work around for now, as I believe we'll have to update the build on CRAN to protect some OpenMP code that was inadvertently exposed and will error on systems that do not support OpenMP.  Another option is to update to Catalina, then Xcode, then move to gcc compilers instead, via Homebrew, or some other package manager, but this is a bit more involved. Either way, you're better off getting a toolchain in place that supports OpenMP and runs this package in parallel rather than serial.

**Comment by hugokitano** [comment]
I tried to install the Cran binary version for Mac OS (https://cran.r-project.org/bin/macosx/el-capitan/contrib/3.6/randomForestSRC_2.9.2.tgz) and I did it from the command line, and it told me that the binary package installed. But when I call library(randomForestSRC) in RStudio, I get the following error:
```
Error: package or namespace load failed for ‘randomForestSRC’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/usr/local/lib/R/3.6/site-library/randomForestSRC/libs/randomForestSRC.so':
  dlopen(/usr/local/lib/R/3.6/site-library/randomForestSRC/libs/randomForestSRC.so, 6): Library not loaded: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libomp.dylib
  Referenced from: /usr/local/lib/R/3.6/site-library/randomForestSRC/libs/randomForestSRC.so
  Reason: image not found
```
I then tried to install it using RStudio, using the "Install Packages" tool and installing it to the two places it recommended. Neither worked, showing the same error. Should I just try the second option then?

**Comment by pat-s** [comment]
Using `gcc` instead of `clang` works for me as a workaround

Install `gcc` (and optionally `ccache`) via `brew install gcc` and put the following into `~/.R/Makevars`:

```
CXX_STD = CXX14

CC=ccache /usr/local/bin/gcc-9
CC11=ccache /usr/local/bin/gcc-9
CC14=ccache /usr/local/bin/gcc-9
CXX=ccache /usr/local/bin/g++-9
CXX11=ccache /usr/local/bin/g++-9
CXX14=ccache /usr/local/bin/g++-9
## -O3 should be faster than -O2 (default) level optimisation ..
CFLAGS=-g -O3 -Wall -pedantic -std=gnu99 -mtune=native -pipe
CXXFLAGS=-g -O3 -Wall -pedantic -std=c++11 -mtune=native -pipe
```

**Comment by kogalur** [comment]
This has been fixed in 2.9.3.  Platforms without an OpenMP capable toolchain should successfully compile and execute the package. Thank you.
